assets/express/src/index.ts

```diff
@@ -1,19 +1,99 @@
-import express from 'express';
+import express, { RequestHandler } from 'express';
 import helmet from 'helmet';
 import cors from 'cors';
 import compression from 'compression';
+import { createHttpTerminator } from 'http-terminator';
 
-const port = process.env.PORT;
+// Environment
+interface Environment {
+  NODE_ENV: 'development' | 'production';
+  PORT: number;
+};
+
+const env: Environment = {
+  NODE_ENV: process.env.NODE_ENV as Environment['NODE_ENV'],
+  PORT: parseInt(process.env.PORT as string, 10),
+};
+
+// Middleware
+const middleware = [
+  helmet(),
+  cors(),
+  compression(),
+];
+
+// Routes
+interface Route {
+  method: 'all' | 'get' | 'post' | 'put' | 'delete' | 'patch' | 'options' | 'head';
+  path: string;
```

Comment:
should we add a `middlewares` property for route-specific middlewares?

Reply:
We could, but there was no need to do so till this point, so I haven't